tools/resourcedocsgen/pkg/docs/utils.go

```diff
@@ -158,3 +159,24 @@ type noCopy struct{}
 
 // Lock is a type hint for go vet. It need not be called.
 func (*noCopy) Lock() {}
+
+// Convert a markdown language tag to a [language.Language] if there is a corresponding
+// language. If no language exists, then _, false will be returned.
+func convertMarkdownLanguage(s string) (language.Language, bool) {
```

Comment:
I feel like I'd stick this in `language` as something like `FromMarkdownLanguage` maybe

Reply:
I'd rather keep `package language` concerned only with defining languages (a tightly constrained and simple task), and keep `convertMarkdownLanguage` in `package docs`. How to convert from markdown tags to a Pulumi language isn't 100% obvious. 

I won't change the behavior of `convertMarkdownLanguage` since this is a refactor PR, but it seems very plausible that `convertMarkdownLanguage("javascript")` should return `(language.NodeJS, true)`. It doesn't now.